### ISSUE TITLE
Firebase Dynamic Link not previewing correctly

### ISSUE THREAD

**Issue by iqfareez** [Amitbhave/flutter-simple-url-preview#19]
I have long url, say **https://flutree.web.app/NWfLo** and I shorten it, adding social metadata in app and it became **https://flut.page.link/hcpa**.

So, when tap the shorten link, it should redirect to the long link.

The problem is, this package returns a preview of the redirected link, not the Firebase Dynamic Link one. 

My Code:
```dart
SimpleUrlPreview(
      url: 'https://flut.page.link/hcpa',
      bgColor: Colors.white,
),
```

Screenshot:
This package: 
![image](https://user-images.githubusercontent.com/60868965/127966392-2cfee174-c49c-4158-b0da-09c654051da4.png)

How I want it to look like:
![image](https://user-images.githubusercontent.com/60868965/127966606-483bb6b2-910b-4230-81b1-e6f330212764.png)

**Comment by Amitbhave** [comment]
@iqfareez yes, that is because we make HTTP call to the URL. Now as that URL redirects it to resultant URL, you will be shown resultant URL's preview..

**Comment by iqfareez** [comment]
I see. Understood but you have any idea I can get the preview of the dynamic link itself?

**Comment by Amitbhave** [comment]
@iqfareez can you tell me more about the requirement here? The screenshot that you have shared, is it the dynamic link's preview shown in whatsapp?

**Comment by iqfareez** [comment]
@Amitbhave I'm not really sure about the requirement. The docs not really helpful. The link preview (that I want) works on major social platform (WhatsApp, Twitter etc).

Dynamic Link social preview doc: https://firebase.google.com/docs/dynamic-links/link-previews
